### PR TITLE
TSK-345 Clarify usage of authorization terms in API

### PR DESCRIPTION
**Pull request by BerndBreier**

### CODE REVIEW

File: lib/taskana-core/src/main/java/pro/taskana/impl/TaskanaEngineImpl.java [hunk 1]
```diff
@@ -356,14 +356,17 @@ protected void initRoles(String propertiesFilename, String propertiesSeparator)
...
                     InputStream inputStream = this.getClass().getResourceAsStream(propertiesFile);
                     if (inputStream == null) {
-                        LOGGER.error("properties file {} was not found on classpath.", propertiesFile);
+                        LOGGER.error("properties file {} for Role configuration was not found on classpath.",
+                            propertiesFile);
                         ensureRoleMapIsFullyInitialized();
                         return;
                     } else {
                         props.load(new InputStreamReader(inputStream));
+                        LOGGER.info("Role properties were loaded from file {} from classpath.", propertiesFile);
```
Comment (MLengl) [inline lib/taskana-core/src/main/java/pro/taskana/impl/TaskanaEngineImpl.java]: loglvl should be DEBUG in my optinion. 
Important logs wouldn´t be unseen if a to high amount of logentries. 

If you agree with this please have a look at all loglevels on this PR.

File: lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java [hunk 2]
```diff
@@ -181,7 +182,7 @@ public WorkbasketAccessItem newWorkbasketAccessItem(String workbasketId, String
     }
 
     @Override
-    public WorkbasketAccessItem createWorkbasketAuthorization(WorkbasketAccessItem workbasketAccessItem)
+    public WorkbasketAccessItem createWorkbasketAccessItem(WorkbasketAccessItem workbasketAccessItem)
         throws InvalidArgumentException, NotAuthorizedException {
         LOGGER.debug("entry to createWorkbasketAuthorization(workbasketAccessItem = {})", workbasketAccessItem);
```
Comment (MLengl) [inline lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java]: Method name changed

File: lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java [hunk 3]
```diff
@@ -246,7 +247,7 @@ public void setWorkbasketAuthorizations(String workbasketId, List<WorkbasketAcce
     }
 
     @Override
-    public void deleteWorkbasketAuthorization(String accessItemId) throws NotAuthorizedException {
+    public void deleteWorkbasketAccessItem(String accessItemId) throws NotAuthorizedException {
         LOGGER.debug("entry to deleteWorkbasketAuthorization(id = {})", accessItemId);
```
Comment (MLengl) [inline lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java]: Method name changed

File: lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java [hunk 4]
```diff
@@ -282,7 +283,7 @@ public void checkAuthorization(String workbasketId,
 
     @Override
     public void checkAuthorization(String workbasketKey, String domain,
-        WorkbasketAuthorization workbasketAuthorization)
+        WorkbasketPermission workbasketAuthorization)
```
Comment (MLengl) [inline lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java]: Pls change parameter/variable name

File: lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java [hunk 5]
```diff
@@ -291,7 +292,7 @@ public void checkAuthorization(String workbasketKey, String domain,
     }
 
     @Override
-    public WorkbasketAccessItem updateWorkbasketAuthorization(WorkbasketAccessItem workbasketAccessItem)
+    public WorkbasketAccessItem updateWorkbasketAccessItem(WorkbasketAccessItem workbasketAccessItem)
         throws InvalidArgumentException, NotAuthorizedException {
         LOGGER.debug("entry to updateWorkbasketAuthorization(workbasketAccessItem = {}", workbasketAccessItem);
```
Comment (MLengl) [inline lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java]: Method name changed

File: lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java [hunk 6]
```diff
@@ -273,7 +274,7 @@ public void deleteWorkbasketAuthorizationForAccessId(String accessId) {
 
     @Override
     public void checkAuthorization(String workbasketId,
```
Comment (MLengl) [inline lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java]: Maybe this Method should be called "checkPermissions"

File: lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java [hunk 7]
```diff
@@ -680,7 +681,7 @@ public WorkbasketAccessItemQuery createWorkbasketAccessItemQuery() throws NotAut
     }
 
     private void checkAuthorization(String workbasketKey, String domain, String workbasketId,
-        WorkbasketAuthorization workbasketAuthorization)
+        WorkbasketPermission workbasketAuthorization)
```
Comment (MLengl) [inline lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java]: Change variable name, too.

File: lib/taskana-core/src/main/java/pro/taskana/mappings/WorkbasketMapper.java [hunk 8]
```diff
@@ -156,7 +156,7 @@
         @Result(property = "orgLevel2", column = "ORG_LEVEL_2"),
         @Result(property = "orgLevel3", column = "ORG_LEVEL_3"),
         @Result(property = "orgLevel4", column = "ORG_LEVEL_4")})
-    List<WorkbasketSummaryImpl> findByPermission(@Param("authorizations") List<WorkbasketAuthorization> authorizations,
+    List<WorkbasketSummaryImpl> findByPermission(@Param("authorizations") List<WorkbasketPermission> authorizations,
```
Comment (MLengl) [inline lib/taskana-core/src/main/java/pro/taskana/mappings/WorkbasketMapper.java]: authorizations should be changed to permissions

File: lib/taskana-core/src/test/java/pro/taskana/impl/WorkbasketServiceImplTest.java [hunk 9]
```diff
@@ -84,7 +85,7 @@ public void testGetWorkbasketById_NonAuthorizedUser()
         throws WorkbasketNotFoundException, NotAuthorizedException {
         String wbId = "ID-1";
         Workbasket wb = createTestWorkbasket(wbId, "Key-1");
-        WorkbasketAuthorization authorization = WorkbasketAuthorization.READ;
+        WorkbasketPermission authorization = WorkbasketPermission.READ;
```
Comment (MLengl) [inline lib/taskana-core/src/test/java/pro/taskana/impl/WorkbasketServiceImplTest.java]: variable name "permission" - multiple times in this class.

File: rest/src/main/java/pro/taskana/rest/resource/mapper/WorkbasketAccessItemMapper.java [hunk 10]
```diff
@@ -43,20 +43,20 @@ public WorkbasketAccessItem toModel(WorkbasketAccessItemResource wbAccItemRecour
     WorkbasketAccessItemResource addLinks(WorkbasketAccessItemResource resource, WorkbasketAccessItem wbAccItem)
         throws NotAuthorizedException {
         resource.add(
-            linkTo(methodOn(WorkbasketController.class).getWorkbasketAuthorizations(wbAccItem.getWorkbasketId()))
+            linkTo(methodOn(WorkbasketController.class).getWorkbasketAccessItems(wbAccItem.getWorkbasketId()))
```
Comment (MLengl) [inline rest/src/main/java/pro/taskana/rest/resource/mapper/WorkbasketAccessItemMapper.java]: withRel("") - names for the Method should be changed, too pls.

File: lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java (hunk 6 above)
Comment (BerndBreier) [reply lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java]: Agreed with Holger to leave the name checkAuthorization

**Pull request merged**